Automattic/pocket-casts-android
# Remove deprecated SubscriptionType type

## Description

This PR refactors the subscription-related code:

- Removed the deprecated `SubscriptionType` type.
- Cleaned up `SubscriptionStatus`:
  - Converted from a `sealed class` to a `sealed interface`.
  - Removed redundant `expiry` and `expiryDate` fields.
  - Removed the subscription list for the free status.

Context: 

## Testing Instructions

1. Verify that different subscription types are displayed correctly in the profile tab and the account details screen using the Admin panel.  
2. Test with the release variant to ensure subscriptions display correctly after performing test purchases.

## Checklist
